packages/auth/src/auth.interfaces.ts

```diff
@@ -7,17 +7,34 @@ export enum AccessMode {
   Offline = 'offline',
 }
 
-export interface ShopifyAuthModuleOptions {
+export type ShopifyAuthModuleAuthorizationCodeFlowBaseOptions = {
   basePath?: string;
   returnHeaders?: boolean;
   useGlobalPrefix?: boolean;
   afterAuthHandler?: ShopifyAuthAfterHandler;
-}
+};
```

Comment:
I would prefer not splitting up the options interface and to be closer to how Shopify/shopify-app-js repo does it: by using a boolean flag.

We can call it `newEmbeddedAuthStrategy?: boolean`

Reply:
What do you mena by "splitting up the options interface"?
Splitting up between the auth code flow and token exchange?
The reason we need two types is that the after auth handler interface is different and I wanted to make it typesafe.
So if you specify `authStrategy: 'TOKEN_FLOW'`, your after auth handler needs to implement `ShopifyAuthTokenExchangeAfterHandler`, otherwise you'd get a type error.